src/DGtal/shapes/DigitalShapesDecorator.h

```diff
@@ -50,6 +50,236 @@
+    BOOST_CONCEPT_ASSERT (( concepts::CDigitalOrientedShape< ShapeA > ));
+
+    typedef typename ShapeA::Space Space;
+    typedef typename ShapeA::Point Point;
+
+    DigitalShapesCSG( )
+    {}
+
+    DigitalShapesCSG ( const DigitalShapesCSG & other )
+      : myShapeA(other.myShapeA), v_shapes(other.v_shapes),
+        myLowerBound(other.myLowerBound), myUpperBound(other.myUpperBound)
+    {}
+
+    DigitalShapesCSG & operator= ( const DigitalShapesCSG & other )
+    {
+      myShapeA = other.myShapeA;
+      v_shapes = other.v_shapes;
+
+      myLowerBound = other.myLowerBound;
+      myUpperBound = other.myUpperBound;
+      return *this;
+    }
+
+    /**
+      * Constructor.
+      *
+      * @param[in] a a model of CDigitalBoundedShape and CDigitalOrientedShape
+      */
+    DigitalShapesCSG( ConstAlias<ShapeA> a )
+      : myShapeA( a )
+    {
+      myLowerBound = myShapeA->getLowerBound();
+      myUpperBound = myShapeA->getUpperBound();
+    }
+
+    /**
+      * Union between a (ShapeA) and b (ShapeB). If an operation was already set, the
+      * union will be between the CSG shape and b (ShapeB).
+      *
+      * @param[in] b a model of CDigitalBoundedShape and CDigitalOrientedShape
+      */
+    void op_union( ConstAlias<ShapeB> b )
+    {
+      BOOST_CONCEPT_ASSERT (( concepts::CDigitalBoundedShape< ShapeB > ));
+      BOOST_CONCEPT_ASSERT (( concepts::CDigitalOrientedShape< ShapeB > ));
+      std::pair<e_operator, CountedConstPtrOrConstPtr< ShapeB > > shape( e_union, b );
+
+      for(uint i =0; i < Space::dimension; ++i)
+      {
+        myLowerBound[i] = std::min(myLowerBound[i], b->getLowerBound()[i]);
+        myUpperBound[i] = std::max(myUpperBound[i], b->getUpperBound()[i]);
+      }
+
+      v_shapes.push_back(shape); 
+    }
+
+    /**
+      * Intersection between a (ShapeA) and b (ShapeB). If an operation was already set, the
+      * intersection will be between the CSG shape and b (ShapeB).
+      *
+      * @param[in] b a model of CDigitalBoundedShape and CDigitalOrientedShape
+      */
+    void op_intersection( ConstAlias<ShapeB> b )
+    {
+      BOOST_CONCEPT_ASSERT (( concepts::CDigitalBoundedShape< ShapeB > ));
+      BOOST_CONCEPT_ASSERT (( concepts::CDigitalOrientedShape< ShapeB > ));
+      std::pair<e_operator, CountedConstPtrOrConstPtr< ShapeB > > shape( e_intersection, b );
+
+      for(uint i=0; i < Space::dimension; ++i)
+      {
+        myLowerBound[i] = std::max(myLowerBound[i], b->getLowerBound()[i]);
+        myUpperBound[i] = std::min(myUpperBound[i], b->getUpperBound()[i]);
+      }
+
+      v_shapes.push_back(shape); 
+    }
+
+    /**
+      * Minus between a (ShapeA) and b (ShapeB). If an operation was already set, the
+      * minus will be between the CSG shape and b (ShapeB).
+      *
+      * @param[in] b a model of CDigitalBoundedShape and CDigitalOrientedShape
+      */
+    void op_minus( ConstAlias<ShapeB> b )
```

Comment:
Not operator-() because we can't represente the intersection operation like this, and the name convention will become inconscistent.